src/hdmint/mintpool.cpp

```diff
@@ -7,109 +7,55 @@
-{
-    Add(make_pair(seedId, nCount));
-    LogPrintf("%s : add %s to mint pool, nCountLastGenerated=%d\n", __func__, seedId.GetHex(), nCountLastGenerated);
-}
-
-void CMintPool::Add(const pair<CKeyID, uint32_t>& pMint, bool fVerbose)
+void CMintPool::Add(pair<uint256, MintPoolEntry> pMint, bool fVerbose)
 {
     insert(pMint);
-    if (pMint.second >= nCountLastGenerated)
-        nCountLastGenerated = pMint.second;
 
     if (fVerbose)
-        LogPrintf("%s : add %s count %d to mint pool\n", __func__, pMint.first.GetHex().substr(0, 6), pMint.second);
+        LogPrintf("%s : add %s count %d to mint pool\n", __func__, pMint.first.GetHex().substr(0, 6), get<2>(pMint.second));
 }
 
-bool CMintPool::Has(const CKeyID& seedId)
+bool SortSmallest(const pair<uint256, MintPoolEntry>& a, const pair<uint256, MintPoolEntry>& b)
 {
-    return static_cast<bool>(count(seedId));
+    return get<2>(a.second) < get<2>(b.second);
 }
 
-bool CMintPool::Get(const CKeyID& seedId, std::pair<CKeyID, uint32_t>& result)
+void CMintPool::List(list<pair<uint256, MintPoolEntry>>& listMints)
 {
-    // for(std::list<pair<CKeyID, uint32_t>>::iterator it = List().begin(); it != List().end(); ++it){
-    //     if((*it).first==seedId){
-    //         result = *it;
-    //         return true;
-    //     }
-    // }
     for (auto pMint : *(this)) {
-        if(pMint.first==seedId){
-           result = pMint;
-           return true; 
-        }
+        listMints.emplace_back(pMint);
     }
-    return false;
+
+    listMints.sort(SortSmallest);
 }
 
-bool CMintPool::Get(const uint32_t& nCount, std::pair<CKeyID, uint32_t>& result)
+void CMintPool::Reset()
 {
-    // std::list<pair<CKeyID, uint32_t> > listMints = List();
-    // for(std::list<pair<CKeyID, uint32_t>>::iterator it = listMints.begin(); it != listMints.end(); ++it){
-    //     if((*it).second==nCount){
-    //         result = *it;
-    //         return true;
-    //     }
-    // }
+    clear();
+}
 
+bool CMintPool::Get(int32_t nCount, uint160 hashSeedMaster, pair<uint256, MintPoolEntry>& result){
     for (auto pMint : *(this)) {
-        if(pMint.second==nCount){
+        if(get<0>(pMint.second)==hashSeedMaster && get<2>(pMint.second)==nCount){
            result = pMint;
-           return true; 
+           return true;
         }
     }
 
     return false;
-}
-
-bool SortSmallest(const pair<CKeyID, uint32_t>& a, const pair<CKeyID, uint32_t>& b)
-{
-    return a.second < b.second;
-}
-
-std::list<pair<CKeyID, uint32_t> > CMintPool::List()
-{
-    list<pair<CKeyID, uint32_t> > listMints;
-    for (auto pMint : *(this)) {
-        listMints.emplace_back(pMint);
-    }
 
-    listMints.sort(SortSmallest);
-
-    return listMints;
 }
-
-void CMintPool::Reset()
-{
-    clear();
-    nCountLastGenerated = 0;
-    nCountLastRemoved = 0;
-}
-
-bool CMintPool::Front(std::pair<CKeyID, uint32_t>& pMint)
+bool CMintPool::Front(pair<uint256, MintPoolEntry> pMint)
```

Comment:
I could not find any use of this. If it is necessary, than we need to pass by reference I think.